File: D01/d01.rb
inputs = File.read('in.txt').split("\n")

# p inputs
calories = []
start = 0
tmp_sum = 0
inputs.each_with_index do |item, i|
  if inputs[i] == ''
    calories.append(tmp_sum)
    tmp_sum = 0
  else
    tmp_sum += inputs[i].to_i
  end
  calories.append(tmp_sum) if tmp_sum != 0 and i == inputs.size - 1
end

# p calories
puts "part 1: #{calories.max}"

puts "part 2: #{calories.sort{ |a, b| a <=> b }.last(3).sum}"

File: D01/d01.py
with open('in.txt') as data:
    inputs = [line.strip() for line in data.readlines()]
# print(inputs)

calories = []
tmp_sum = 0
last_index =  len(inputs) - 1
for i_ in range(len(inputs)):
    if inputs[i_] == "":
        calories.append(tmp_sum)
        tmp_sum = 0
    else:
        # print(inputs[i_])
        tmp_sum += int(inputs[i_])
    if i_ == last_index and tmp_sum != 0:
        calories.append(tmp_sum)

# print(calories)
print(f"part 1: {max(calories)}")

# print(sorted(calories)[-3:])
print(f"part 2: {sum(sorted(calories)[-3:])}")
